connectivity/check/wait.go

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package check
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium-cli/defaults"
+	"github.com/cilium/cilium-cli/k8s"
+)
+
+const (
+	LongTimeout  = 5 * time.Minute
+	ShortTimeout = 30 * time.Second
+
+	PollInterval = 1 * time.Second
+)
+
+// WaitForDeployment waits until the specified deployment becomes ready.
+func WaitForDeployment(ctx context.Context, log Logger, client *k8s.Client, namespace string, name string) error {
+	log.Logf("⌛ [%s] Waiting for deployment %s/%s to become ready...", client.ClusterName(), namespace, name)
```

Comment:
This is a common pattern across all subsequent waitFor calls. It seems it might be a  prime candidates for abstracting out these calls. 
Also do these need to be exported?

Reply:
Thanks Marco. I see. Figured this is common logic and best to abstract with a closure. 
Personally I am not keen about making logging decisions deep in the stack as it should be the responsibility of the call site to determine whether to handle the error, log it, etc... If potentially multi errors can be surfaced then we should return these accordingly. If this is outside the scope of this review, I am good with that but figured I'll flag.

Reply:
I appreciate the clarification and I see your point. I'd personally not change it further in this PR, to avoid modifying the pre-existing behavior. Especially considering the logs, given that the CLI used to output a debug log for every retry in case of errors.